tests/dataloader_tests.cpp

```diff
@@ -0,0 +1,33 @@
+/**
+ * @file dataloader_tests.cpp
+ * @author Kartik Dutt
+ *
+ * Tests for various functionalities of dataloader.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#define BOOST_TEST_DYN_LINK
+#define BOOST_TEST_MAIN // Do not define this anywhere else.
+#include <dataloader/dataloader.hpp>
+#include <boost/test/unit_test.hpp>
+using namespace boost::unit_test;
+
+BOOST_AUTO_TEST_SUITE(DataLoadersTest);
+
+/**
+ * Simple test for MNIST Dataloader.
+ */
+BOOST_AUTO_TEST_CASE(MNISTDataLoaderTest)
+{
+  DataLoader<> dataloader("mnist", true, 0.80);
+  // Check for correct dimensions.
+  BOOST_REQUIRE_EQUAL(dataloader.TrainX().n_cols, 784);
+  BOOST_REQUIRE_EQUAL(dataloader.TestX().n_cols, 784);
+  BOOST_REQUIRE_EQUAL(dataloader.ValidX().n_cols, 784);
+  BOOST_REQUIRE_EQUAL(dataloader.TrainX().n_rows, 33600);
+}
```

Comment:
Want to add some tests for when `DataLoader` is used incorrectly?  (i.e., does it crash as expected?)

Reply:
Agreed, I will add them as well.